typings/objection/index.d.ts

```diff
@@ -1343,7 +1343,26 @@ declare namespace Objection {
     rollback<QM>(error?: Error): Promise<QM>;
   }
 
-  // The following is from https://gist.github.com/enriched/c84a2a99f886654149908091a3183e15
+  type JsonSchemaFormatType =
+    | 'date-time'
+    | 'time'
+    | 'date'
+    | 'email'
+    | 'idn-email'
```

Comment:
So.. this one is a bit tricky. Objection uses `ajv`, which doesn't actually support `idn-email` despite it being an option in the spec for json schema.

Depending on where/how this JsonSchema type gets used, in my opinion it'd be better to leave out options that don't work/aren't supported in this context. I'm also not a regular contributor to this project though, so my opinions don't hold a lot of weight around here.

Reply:
you can replace ajv with your own JsonSchema validator and overriding this type for your custom models is impossible-same as with custom QueryBuilders. So I would leave the whole spec there in place. It's just annoying to use in a TS project if you need to litter `@ts-ignore` all over your objection models.

Reply:
solid point. agreed, it makes sense to keep it in this case.